plugin/pkg/admission/network/defaultingressclass/admission_test.go

```diff
@@ -196,3 +196,220 @@ func TestAdmission(t *testing.T) {
 		})
 	}
 }
+
+func TestAdmissionUpdated(t *testing.T) {
+	defaultClass1 := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default1",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "true",
+			},
+		},
+	}
+	defaultClass2 := &networkingv1.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default2",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "true",
+			},
+		},
+	}
+	// Class that has explicit default = false
+	classWithFalseDefault := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nondefault1",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "false",
+			},
+		},
+	}
+	// Class with missing default annotation (=non-default)
+	classWithNoDefault := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nondefault2",
+		},
+	}
+	// Class with empty default annotation (=non-default)
+	classWithEmptyDefault := &networkingv1.IngressClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IngressClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nondefault2",
+			Annotations: map[string]string{
+				networkingv1.AnnotationIsDefaultIngressClass: "",
+			},
+		},
+	}
+
+	testCases := []struct {
+		name                  string
+		classesBefore         []*networkingv1.IngressClass
+		classesAfter          []*networkingv1.IngressClass
+		classFieldBefore      *string
+		classFieldAfter       *string
+		classAnnotationBefore *string
+		classAnnotationAfter  *string
+		expectedClass         *string
+		expectedError         error
+	}{
+		{
+			name:                  "no default, no modification of Ingress",
+			classesBefore:         []*networkingv1.IngressClass{classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			classesAfter:          []*networkingv1.IngressClass{classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
+			classFieldBefore:      nil,
+			classFieldAfter:       nil,
+			classAnnotationBefore: nil,
+			classAnnotationAfter:  nil,
```

Comment:
I may be missing something, but it looks like these fields are never actually set in the test cases. I want to be sure this change interacts well with the annotation, otherwise this PR LGTM.

Reply:
If I understand correctly, are you referring to this test case alone?
This is to test the situation when there is no default value, but it may not be necessary.

I am now modifying this test case as follows: there is a default ingress class before, and there is no default ingress class later, to test this situation.

Reply:
Sorry for the confusion, my concern was primarily that it didn't seem like `classAnnotationBefore` and `classAnnotationAfter` were being used by any test cases. It looks like `classAnnotationAfter` is used now, but I'm not seeing `classAnnotationBefore` specified anywhere.

Reply:
This place call classAnnotationBefore:
https://github.com/kubernetes/kubernetes/pull/107018/files#diff-dbe2c160789adc3a41cf1f1157fe5733ecbf1c19fb15922e9747b719c1b18ca0R381

But the test cases do lack examples of using `classAnnotationBefore`, do you mean that?

Reply:
Yep I noticed that there was code to implement it if it was set by any test cases, I just didn't see a non nil value for that field in any test cases.

Reply:
I know what you mean. This is my negligence. I accidentally ignored this situation while writing test cases. Thanks！